### PR TITLE
[core] Attempt constraining Y axis ratio first, then X axis

### PR DESCRIPTION
**Pull request by brunoabinader**
This matches GL JS default constraint for the map boundaries. Parity ensured via newly added `{basic,bright,satellite}-v9/z0-wide` render tests.

### CODE REVIEW

**Comment by brunoabinader** [comment]
@mourner even though GL JS and Native are now in sync with regards to the transform constraint results, the heatmap produced by GL native is now different - it actually looks more correct as the color ramp is more pronounced - could you please check these render tests results and check - 
 ( https://141939-16286131-gh.circle-artifacts.com/0/render-tests )? My suspicion is that this has shown a bug in GL JS side.

**Comment by mourner** [comment]
Not sure what's exactly going on but it might be a bug in GL Native — the heatmap is more intense likely because there are two points (original and duplicate from another tile) on top of each other.

**Comment by brunoabinader** [comment]
> Not sure what's exactly going on but it might be a bug in GL Native — the heatmap is more intense likely because there are two points (original and duplicate from another tile) on top of each other.

I've revisited this and found out that GL JS uses the following logic: Attempt to constrain the Y axis if its scale (height / tile size) is bigger than 1 - if not, attempt constrain the X axis instead.

Updating this new logic in GL Native fixes the render test results. @mourner could you please re-review?

File: test/style/expression/expression.test.cpp [hunk 1]
```diff
@@ -29,18 +29,21 @@ TEST(Expression, IsExpression) {
...
+        // TODO: "interpolate-hcl": https://github.com/mapbox/mapbox-gl-native/issues/8720
+        // TODO: "interpolate-lab": https://github.com/mapbox/mapbox-gl-native/issues/8720
+        // TODO:          "format": https://github.com/mapbox/mapbox-gl-native/issues/12612
+        if (name == "feature-state" || name == "interpolate-hcl" || name == "interpolate-lab" || name == "format") {
+            if (expression::isExpression(conversion::Convertible(expression))) {
+                ASSERT_TRUE(false) << "Expression name" << name << "is implemented - please update Expression.IsExpression test.";
+            }
+            continue;
+        }
+
```
Comment (mourner) [inline test/style/expression/expression.test.cpp]: This doesn't seem related to the PR — perhaps committed accidentally?

Comment (brunoabinader) [reply test/style/expression/expression.test.cpp]: It is - line-progress was implemented already and this new logic helps cleaning up TODOs like this when stuff gets implemented.
